decode/pool.go

```diff
@@ -66,19 +57,10 @@ func (p *Pool) Run() {
 				p.logger.Log("Decoder exiting")
 				return
 			}
+
 		default:
-			err := p.src.DiscardPacket()
-			if err == pcap.NextErrorTimeoutExpired {
-				// loop again
-			} else if err == io.EOF {
-				// wait for a worker to become ready so we can
-				// shut them down and avoid a busy-wait loop.
-				time.Sleep(10 * time.Millisecond)
-			} else if err == nil {
-				p.stats.PacketsDropped++
-			} else {
-				p.logger.Log("Error from DiscardPacket", err)
-			}
+			p.startWorker(p.numWorkers)
+			p.numWorkers++
```

Comment:
optionally, read/modify numWorkers within startWorker. would mean removing the (id int) argument from startWorker, and reading it from p within startWorker, e.g.
id := p.numWorkers
p.numWorkers++